subprojects/docs/src/docsTest/java/org/gradle/docs/samples/IntegrationTestSamplesExecutor.java

```diff
@@ -62,6 +64,27 @@ class IntegrationTestSamplesExecutor extends CommandExecutor {
 
     @Override
     protected int run(String executable, List<String> args, List<String> flags, OutputStream outputStream) {
+        try {
+            if (expectFailure) {
+                // TODO(mlopatkin) sometimes it is still possible to save the configuration cache state in the event of failure.
+                //  We need to figure out how to separate expected failure from the CC store failure.
+                ExecutionFailure result = createExecuter(args, flags).runWithFailure();
+                outputStream.write((result.getOutput() + result.getError()).getBytes());
+            } else {
+                ExecutionResult result = createExecuter(args, flags).run();
+                outputStream.write(result.getOutput().getBytes());
+
+                if (shouldCheckLoadingFromConfigurationCache()) {
+                    createExecuter(args, flags).withArgument("--rerun-tasks").run();
```

Comment:
❓ I wonder what happens to the build output when the 2nd run fails 🤔

Reply:
It is nicely added to the failure message, see, for example, https://ge.gradle.org/s/2x4hcknqwf43s/tests/:docs:docsTest/org.gradle.docs.samples.Bucket3SnippetsTest/snippet-ear-ear-with-war_groovy_earWithWar.sample?top-execution=1